File: 001_two_sum/ts/solution.js
"use strict";
function twoSum(nums, target) {
    const map = new Map();
    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (map.has(complement)) {
            return [map.get(complement), i];
        }
        map.set(nums[i], i);
    }
    return [-1, -1];
}
function main() {
    const aux = [2, 7, 11, 15];
    const result = twoSum(aux, 9);
    for (let i = 0; i < result.length; i++) {
        console.log(result[i]);
    }
}
main();

File: 001_two_sum/solution.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in num_map:
                return [num_map[complement], i]
            num_map[num] = i
        return [-1, -1]

def main():
    aux = [2, 7, 11, 15]
    obj = Solution()
    result = obj.twoSum(aux, 9)
    for i in range(len(result)):
        print(result[i])

if __name__ == "__main__":
    main()